src/test/java/io/confluent/idesidecar/restapi/cache/ClusterCacheTest.java

```diff
@@ -224,6 +225,16 @@ void shouldFindAlreadyLoadedSchemaRegistryForConnectionAndLazilyLoadKafkaCluster
       verifyNoInteractions(local);
     }
 
+    @Test
+    void shouldClearCacheOnConnectionUpdateAndReCacheDetails() {
+      // When a connection is updated
+      cache.onConnectionUpdated(CONNECTION_1);
+
+      // assert cache is empty
+      assertEquals(0, cache.clustersByConnectionId.size());
+
+    }
+
```

Comment:
I think we essentially want to do something like this:
```suggestion
    @Test
    void shouldRemoveCacheOnConnectionUpdate() {
      // When two connections are created
      cache.onConnectionCreated(CONNECTION_1);
      cache.onConnectionCreated(CONNECTION_2);

      // and a Kafka cluster is loaded in each connection
      cache.onLoadingKafkaCluster(
          new ClusterEvent(CONNECTION_1_ID, ConnectionSpec.ConnectionType.CCLOUD, LKC_1)
      );
      cache.onLoadingKafkaCluster(
          new ClusterEvent(CONNECTION_2_ID, ConnectionSpec.ConnectionType.CCLOUD, LKC_2)
      );

      // Then there will be a connection cache for both connections
      assertConnection(CONNECTION_1);
      assertConnection(CONNECTION_2);

      // and the cache for each connection contains their Kafka cluster
      assertKafkaCluster(LKC_1);
      assertKafkaCluster(LKC_2);

      // When one of the connections is updated
      cache.onConnectionUpdated(CONNECTION_1);

      // then the cache for the connection will have been cleared (no cluster will be found)
      var clusters = assertConnection(CONNECTION_1);
      assertThrows(
          ClusterNotFoundException.class,
          () -> clusters.getKafkaCluster(LKC_1.id(), false)
      );

      // but the cache for the other connection is unaffected
      assertConnection(CONNECTION_2);
      assertKafkaCluster(LKC_2);

      // and the Kafka cluster for connection 1 will be loaded again
      expectFindKafkaCluster(CONNECTION_1_ID, LKC_1.id(), LKC_1);
      // when the Kafka cluster is accessed again
      assertEquals(LKC_1, clusters.getKafkaCluster(LKC_1.id()));
    }
```
This does a couple of things. First, it verifies that the cache for a connection is cleared when it is updated. We're not quite checking that it is cleared, but we're verifying that a cluster that used to be there no longer is in the cache. We could add an `isEmpty()` protected method to the `Clusters` class, and also add an assertion in the test that it is empty.

Second, it verifies that other connections are not affected. The cluster that was loaded into the cache for connection 2 is still there after connection 1 is updated.

Third, it verifies that after a connection is updated and it's cache is cleared (or technically recreated), then the cache for connection 1 will reload the Kafka cluster if it is queried.

BTW, here are the additional constants that are needed for `CONNECTION_2` and `LKC_2`:
```
    static final String CONNECTION_2_ID = "c2";
    static final ConnectionState CONNECTION_2 = ConnectionStates.from(
        new ConnectionSpec(
            CONNECTION_2_ID,
            "Conn2",
            ConnectionSpec.ConnectionType.CCLOUD
        ),
        null
    );
    static final CCloudOrganization ORG_2 = new CCloudOrganization(
        "org2",
        "my-other-org",
        false,
        CONNECTION_2_ID
    );
    static final CCloudEnvironment ENV_2 = new CCloudEnvironment(
        "env-456",
        "my-env-2",
        CCloudGovernancePackage.ESSENTIALS
    ).withOrganization(ORG_2).withConnectionId(CONNECTION_2_ID);
    static final CCloudKafkaCluster LKC_2 = new CCloudKafkaCluster(
        "lkc-2",
        "my-other-kafka-cluster",
        CloudProvider.AWS,
        "us-west-2",
        "pkc-456"
    ).withEnvironment(ENV_2).withOrganization(ORG_2).withConnectionId(CONNECTION_2_ID);
```